packages/ui-components/src/components/tab-list/tab-list.tsx

```diff
@@ -57,14 +48,20 @@ export class KvTabList {
 	private calculateSelectionAnimationProperties() {
 		const tabIndex = this.tabItems?.findIndex(tab => tab.tabKey === this.selectedTabKey);
 
-		if (gte(tabIndex, 0) && !isEmpty(this.tabItems)) {
+		if (gte(tabIndex, 0) && !isEmpty(this.tabItems) && !isEmpty(this.selectedTabKey)) {
 			const selectedTabEl = this.tabItems.find((_tabRef, index) => tabIndex === index);
 			const labelEl: HTMLElement = selectedTabEl.shadowRoot.querySelector('.label');
 			this.selectedTabIndicatorOffset = labelEl.offsetLeft;
 			this.selectedTabIndicatorWidth = labelEl.clientWidth / 2;
 		}
 	}
 
+	private handleSlotChange = () => {
```

Comment:
you shouldn't replicate the code please call the `handleSlotChange` inside the `componentDidRender`.